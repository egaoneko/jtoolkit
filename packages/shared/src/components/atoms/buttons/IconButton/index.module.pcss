.button {
  @apply p-2.5;
}
