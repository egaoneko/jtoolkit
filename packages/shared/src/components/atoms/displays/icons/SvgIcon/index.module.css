.svg {
  &.doubleExtraSmall {
    @apply text-xs;
  }

  &.extraSmall {
    @apply text-sm;
  }

  &.small {
    @apply text-lg;
  }

  &.medium {
    @apply text-2xl;
  }

  &.large {
    @apply text-4xl;
  }

  &.extraLarge {
    @apply text-6xl;
  }

  &.doubleExtraLarge {
    @apply text-8xl;
  }

  &.stroked {
    @apply stroke-current;

    *[stroke] {
      @apply stroke-current;
    }
  }

  &.filled {
    @apply fill-current;

    *[fill] {
      @apply fill-current;
    }
  }
}
