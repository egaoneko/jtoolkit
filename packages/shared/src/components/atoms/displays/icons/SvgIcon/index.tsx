import { forwardRef, SVGAttributes } from 'react';
import clsx from 'clsx';

import { Size } from 'src/enums/size';

import styles from './index.module.css';

export interface SvgIconProps extends SVGAttributes<SVGSVGElement> {
  size?: Size;
  stroked?: boolean;
  filled?: boolean;
  viewBox?: string;
  className?: string;
}

const SvgIcon = forwardRef<SVGSVGElement, SvgIconProps>(function SvgIcon(
  { size = Size.MEDIUM, stroked = false, filled = true, viewBox = '0 0 24 24', className, children, ...rest },
  ref,
) {
  return (
    <svg
      ref={ref}
      className={clsx(styles.svg, styles[size], { [styles.stroked]: stroked, [styles.filled]: filled }, className)}
      width="1em"
      height="1em"
      viewBox={viewBox}
      {...rest}
    >
      {children}
    </svg>
  );
});

export default SvgIcon;
