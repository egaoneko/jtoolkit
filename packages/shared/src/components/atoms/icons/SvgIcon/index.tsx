import { FC, SVGAttributes } from 'react';
import clsx from 'clsx';

import { Size } from '../../../../enums/size';

import styles from './index.module.pcss';

export interface SvgIconProps extends SVGAttributes<SVGElement> {
  size?: Size;
  stroked?: boolean;
  filled?: boolean;
  viewBox?: string;
  className?: string;
}

const SvgIcon: FC<SvgIconProps> = ({
  size = Size.MEDIUM,
  stroked = false,
  filled = true,
  viewBox = '0 0 24 24',
  className,
  children,
  ...rest
}) => {
  return (
    <svg
      {...rest}
      className={clsx(styles.svg, styles[size], { [styles.stroked]: stroked }, { [styles.filled]: filled }, className)}
      width="1em"
      height="1em"
      viewBox={viewBox}
    >
      {children}
    </svg>
  );
};

export default SvgIcon;
