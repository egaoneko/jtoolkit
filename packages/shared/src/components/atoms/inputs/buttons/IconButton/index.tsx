import { FC } from 'react';
import clsx from 'clsx';

import Button, { ButtonProps } from 'src/components/atoms/inputs/buttons/Button';

import styles from './index.module.css';

const IconButton: FC<ButtonProps> = ({ className, children, ...props }) => {
  return (
    <Button {...props} rippleCenter={true} className={clsx(styles.button, className)}>
      {children}
    </Button>
  );
};

export default IconButton;
