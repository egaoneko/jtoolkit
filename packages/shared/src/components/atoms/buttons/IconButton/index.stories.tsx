import { Story } from '@storybook/react';

import { Color } from '../../../../enums/color';
import { ButtonVariant } from '../../../../enums/button';
import { Size } from '../../../../enums/size';
import SvgIcon from '../../icons/SvgIcon';
import { ButtonProps } from '../Button';

import IconButton from './index';

export default {
  title: 'atoms/buttons/IconButton',
  component: IconButton,
};

const icon = (
  <SvgIcon>
    <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
  </SvgIcon>
);
const StandardTemplate: Story<ButtonProps> = args => (
  <div className="flex gap-2">
    <IconButton {...args}>{icon}</IconButton>
  </div>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  size: Size.MEDIUM,
  color: Color.PRIMARY,
  variant: ButtonVariant.CONTAINED,
  rounded: false,
};
