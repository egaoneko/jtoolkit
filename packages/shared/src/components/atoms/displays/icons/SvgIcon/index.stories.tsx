import { Story } from '@storybook/react';

import { Size } from 'src/enums/size';
import styles from 'src/styles/storybook.css';

import SvgIcon, { SvgIconProps } from './index';

export default {
  title: 'atoms/displays/icons/SvgIcon',
  component: SvgIcon,
};

const StandardTemplate: Story<SvgIconProps> = args => {
  const icon = <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />;
  return (
    <div className={styles.row}>
      <div className={styles.primary}>
        <SvgIcon {...args}>{icon}</SvgIcon>
      </div>
      <div className={styles.secondary}>
        <SvgIcon {...args}>{icon}</SvgIcon>
      </div>
      <div className={styles.success}>
        <SvgIcon {...args}>{icon}</SvgIcon>
      </div>
      <div className={styles.danger}>
        <SvgIcon {...args}>{icon}</SvgIcon>
      </div>
      <div className={styles.warning}>
        <SvgIcon {...args}>{icon}</SvgIcon>
      </div>
      <div className={styles.info}>
        <SvgIcon {...args}>{icon}</SvgIcon>
      </div>
    </div>
  );
};

export const Standard = StandardTemplate.bind({});

Standard.args = {
  size: Size.MEDIUM,
  stroked: false,
  filled: true,
};
