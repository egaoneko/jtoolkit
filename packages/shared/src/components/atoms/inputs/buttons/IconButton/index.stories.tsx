import { Story } from '@storybook/react';

import { Color } from 'src/enums/color';
import { ButtonVariant } from 'src/enums/button';
import { Size } from 'src/enums/size';
import SvgIcon from 'src/components/atoms/displays/icons/SvgIcon';
import { ButtonProps } from 'src/components/atoms/inputs/buttons/Button';

import IconButton from './index';

export default {
  title: 'atoms/inputs/buttons/IconButton',
  component: IconButton,
};

const icon = (
  <SvgIcon>
    <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
  </SvgIcon>
);
const StandardTemplate: Story<ButtonProps> = args => (
  <div className="flex gap-2">
    <IconButton {...args}>{icon}</IconButton>
  </div>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  size: Size.MEDIUM,
  color: Color.PRIMARY,
  variant: ButtonVariant.CONTAINED,
  rounded: false,
};
