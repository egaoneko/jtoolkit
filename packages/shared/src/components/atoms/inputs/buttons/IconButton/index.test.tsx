import renderer, { ReactTestInstance } from 'react-test-renderer';

import SvgIcon from 'src/components/atoms/displays/icons/SvgIcon';
import IconButton from 'src/components/atoms/inputs/buttons/IconButton/index';

describe('IconButton Snapshots', () => {
  const icon = (
    <SvgIcon>
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
    </SvgIcon>
  );

  test('renders IconButton with default options', async () => {
    const tree = renderer.create(<IconButton>{icon}</IconButton>);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).type).toBe('button');
    expect(
      (
        (
          ((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance)
            .children[0] as ReactTestInstance
        ).children[0] as ReactTestInstance
      ).type,
    ).toContain('svg');
  });

  test('renders IconButton with class', async () => {
    const tree = renderer.create(<IconButton className="test">{icon}</IconButton>);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('test');
  });
});
