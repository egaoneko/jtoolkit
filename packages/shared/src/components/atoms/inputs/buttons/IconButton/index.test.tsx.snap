// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Snapshots renders IconButton with class 1`] = `
<button
  className="button contained medium primary button test"
  disabled={false}
  onClick={[Function]}
>
  <svg
    className="svg medium filled"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
  <span
    className="ripple"
  />
</button>
`;

exports[`IconButton Snapshots renders IconButton with default options 1`] = `
<button
  className="button contained medium primary button"
  disabled={false}
  onClick={[Function]}
>
  <svg
    className="svg medium filled"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
  <span
    className="ripple"
  />
</button>
`;
